src/test/java/uk/gov/laa/gpfd/service/MetadataReportServiceTest.java

```diff
@@ -0,0 +1,254 @@
+    }
+
+    @Test
+    void shouldHandleConcurrentRequests() {
+        // Given
+        var validId1 = 5;
+        var validId2 = 6;
+        var reportDetails1 = new ReportsGet200ResponseReportListInner() {{
+            setId(validId1);
+            setReportName("Test Report 1");
+        }};
+        var reportDetails2 = new ReportsGet200ResponseReportListInner() {{
+            setId(validId2);
+            setReportName("Test Report 2");
+        }};
+        when(mappingTableService.getDetailsForSpecificReport(validId1)).thenReturn(reportDetails1);
+        when(mappingTableService.getDetailsForSpecificReport(validId2)).thenReturn(reportDetails2);
+
+        // When
+        var response1 = reportService.createReportResponse(validId1);
+        var response2 = reportService.createReportResponse(validId2);
+
+        // Then
+        assertNotEquals(response1.getId(), response2.getId(), "Concurrent requests should produce distinct responses");
+        assertNotEquals(response1.getReportName(), response2.getReportName(), "Concurrent requests should produce distinct responses");
+    }
+
+    @Test
+    void shouldLogErrorWhenReportNotFound() {
+        // Given
+        var missingId = 100;
+        when(mappingTableService.getDetailsForSpecificReport(missingId)).thenThrow(new ReportIdNotFoundException("Report not found"));
+
+        // When
+        // Then
+        assertThrows(ReportIdNotFoundException.class, () -> reportService.createReportResponse(missingId), "Should log an error for a missing report");
+    }
+
+    @Test
+    void shouldReturnResponseForBoundaryId() {
+        // Given
+        var boundaryId = 1;
+        var reportDetails = new ReportsGet200ResponseReportListInner() {{
+            setId(boundaryId);
+            setReportName("Boundary Report");
+        }};
+        when(mappingTableService.getDetailsForSpecificReport(boundaryId)).thenReturn(reportDetails);
+
+        // When
+        var response = reportService.createReportResponse(boundaryId);
+
+        // Then
+        assertEquals(boundaryId, response.getId(), "Boundary ID should match");
+        assertEquals("Boundary Report", response.getReportName(), "Boundary report name should match");
+    }
+
+    @Test
+    void shouldHandleNullReportName() {
+        // Given
+        var validId = 7;
+        var reportDetails = new ReportsGet200ResponseReportListInner() {{
+            setId(validId);
+            setReportName(null);
+        }};
+        when(mappingTableService.getDetailsForSpecificReport(validId)).thenReturn(reportDetails);
+
+        // When
+        var response = reportService.createReportResponse(validId);
+
+        // Then
+        assertNull(response.getReportName(), "Report name should be null");
+    }
+
+    @Test
+    void shouldHandleNullUrlFieldGracefully() {
```

Comment:
Url is not added to any of the above tests, this test is this the same as shouldCreateReportResponseSuccessfully